src/bindings/python/flux/uri/resolvers/lsf.py

```diff
@@ -0,0 +1,102 @@
+###############################################################
+# Copyright 2022 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+
+import re
+import subprocess
+from pathlib import PurePath
+
+from flux.uri import FluxURIResolver, URIResolverPlugin
+
+
+"""
+This module supports the Flux URI resolver for IBM Spectrum LSF.
+
+It can be used like:
+  `flux uri --remote lsf:<JOBID>`.
+
+"""
+
+
+def lsf_find_compute_node(jobid) -> str:
+    """Figure out where the job is being run using JSON output from `bjobs -l`."""
+
+    """
+    To get the name of the system (lassen or sierra or nid) without the node number,
+    pull the hostname and strip off anything that's not alphabetic.
+    """
+    host_system = subprocess.run(["hostname"], capture_output=True).stdout.decode(
+        "utf-8"
+    )
```

Comment:
I would second the ``socket.gethostname`` comment. If you find yourself ever running a process to do some simple thing in Python, do a web search first to make sure there isn't a function somewhere in the standard library to do it for you. I don't think I have ever needed to run processes simply to replicate UNIX commands.

Reply:
So, while thinking about this last night, I wondered if there was a python library that could support SSH. Turns out, there is (https://www.paramiko.org/) but I don't think its worth adding to the python modules used in Flux. And I'm not sure it would play nice with BlueOS systems, so definitely not worth the trouble. But this is a good note about best practices in Python (and general) development, thanks @jameshcorbett and @grondo.